src/Form/index.tsx

```diff
@@ -589,6 +677,24 @@ function Form({
             };
           if (res.save_url_params) saveUrlParamsFormSetting = true;
           setFormSettings(mapFormSettingsResponse(res));
+          setLogicRules(res.logic_rules);
+
+          // Add any logic_rule.elements to viewElements so that onView called for then too.
+          // Make sure there are no duplicate entries.
+          if (res.logic_rules) {
+            const newViewElements: string[] = [...viewElements];
```

Comment:
IMO we should handle this viewElements aggregation on the backend before returning to frontend

Reply:
This is aggregating any view rule elements with any onViewElements that might have been passed as a form prop (embedded situation).  Doing this aggregation on the BE means that onViewElements would need to be sent to the BE via a new end point.  Unless I am missing something, that would not make sense.